README.rst

```diff
@@ -0,0 +1,419 @@
+
+    for epoch in range(100):
+       for batch_idx, (data, target) in enumerate(train_loader):
+           optimizer.zero_grad()
+           output = model(data)
+           loss = F.nll_loss(output, target)
+           loss.backward()
+           optimizer.step()
+           if batch_idx % args.log_interval == 0:
+               print('Train Epoch: {} [{}/{}]\tLoss: {}'.format(
+                   epoch, batch_idx * len(data), len(train_sampler), loss.item()))
+
+
+**Note**: PyTorch support requires NCCL 2.2 or later. It also works with NCCL 2.1.15 if you are not using RoCE or InfiniBand.
+
+mpi4py
+------
+Horovod supports mixing and matching Horovod collectives with other MPI libraries, such as `mpi4py <https://mpi4py.scipy.org>`_,
+provided that the MPI was built with multi-threading support.
+
+You can check for MPI multi-threading support by querying the ``hvd.mpi_threads_supported()`` function.
+
+.. code-block:: bash
+
+    import horovod.tensorflow as hvd
+
+    # Initialize Horovod
+    hvd.init()
+
+    # Verify that MPI multi-threading is supported.
+    assert hvd.mpi_threads_supported()
+
+    from mpi4py import MPI
+    assert hvd.size() == MPI.COMM_WORLD.Get_size()
+
+
+Inference
+---------
+Learn how to optimize your model for inference and remove Horovod operations from the graph `here <docs/inference.md>`_.
+
+
+Tensor Fusion
+-------------
+One of the unique things about Horovod is its ability to interleave communication and computation coupled with the ability
+to batch small *allreduce* operations, which results in improved performance. We call this batching feature Tensor Fusion.
+
+See `here <docs/tensor-fusion.md>`__ for full details and tweaking instructions.
+
+
+Analyzing Horovod Performance
+-----------------------------
+Horovod has the ability to record the timeline of its activity, called Horovod Timeline.
+
+.. image:: https://user-images.githubusercontent.com/16640218/29735271-9e148da0-89ac-11e7-9ae0-11d7a099ac89.png
+   :alt: Horovod Timeline
+
+See `here <docs/timeline.md>`__ for full details and usage instructions.
+
+
+Guides
+------
+Run distributed training in Microsoft Azure using `Batch AI and Horovod <https://github.com/Azure/BatchAI/tree/master/recipes/Horovod>`_.
+
+Troubleshooting
+---------------
+See the `Troubleshooting <docs/troubleshooting.md>`_ page and please submit a `ticket <https://github.com/uber/horovod/issues/new>`_
+if you can't find an answer.
+
+
+Citation
+--------
+Please cite Horovod in your publications if it helps your research:
+
+.. code-block:: bash
+
+    @article{sergeev2018horovod,
+      Author = {Alexander Sergeev and Mike Del Balso},
+      Journal = {arXiv preprint arXiv:1802.05799},
+      Title = {Horovod: fast and easy distributed deep learning in {TensorFlow}},
+      Year = {2018}
+    }
+
+
+Publications
+------------
+1. Sergeev, A., Del Balso, M. (2017) *Meet Horovod: Uber’s Open Source Distributed Deep Learning Framework for TensorFlow*.
```

Comment:
This list doesn't render correctly.

Reply:
May be caused by the line breaks between numbers

Reply:
For some reason, I had to add `Retrieved from`, but now it works